simapp/simd/cmd/root.go

```diff
@@ -101,7 +102,7 @@ func NewRootCmd() *cobra.Command {
 		},
 	}
 
-	initRootCmd(rootCmd, encodingConfig)
+	initRootCmd(rootCmd, encodingConfig, tempApp.BasicModuleManager)
```

Comment:
this feels big bad

Reply:
Why do we need this tempApp and where is this coming from?

Reply:
The origin is not to use depinject for app v1 (as this means we do not show how to get those values for app v1 apps).
For app v2 app, we can just use depinject. I can remove it to clean it up, but then it does mean that when building with the app legacy flag, it will still use the app wiring config.

Reply:
One thing I can simply do is try to create a root.go and root_v2.go for both implementations. So we can simplify the example for those using depinject, where it is way cleaner.

Reply:
Yeah that'd be fire actually!

Reply:
Done, for app v1 it does need to stay like this. app v2 is way cleaner :)